### PR TITLE
Bug 875062: Social Sharing

### PR DESCRIPTION
**Pull request by stephaniehobson**
Includes HTML, CSS, and JS for social sharing buttons.

Now with URL shortening. Had to move URL shortening stuff out of demos and into core.

This probably needs tests now.

Supersedes  #3117.

### CODE REVIEW

**Comment by darkwing** [comment]
URL shortening?

**Comment by stephaniehobson** [comment]
@darkwing We wanted to use mzl.la URL shortening for click counting, to get an idea of how much traffic was coming back to MDN through links shared from the buttons.

**Comment by stephaniehobson** [comment]
But it's not configured, I asked on the bug if it's okay to go forward without it.

File: media/js/social.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+(function(win, doc, $) {
+
+    $('.share-link').on('click', function(e) {
+        e.preventDefault();
+
+        var $this = $(this);
+        var title = encodeURIComponent($this.data('title') || doc.title);
+        var url = encodeURIComponent($this.data('url') || win.location);
```
Comment (groovecoder) [inline media/js/social.js]: We should add `utm_*` tracking parameters for this so we can segment and analyze visitors who come from this feature. I suggest:
`utm_campaign=share&utm_source=[twitter.com|facebook.com|plus.google.com]&utm_medium=share widget`

Comment (stephaniehobson) [inline media/js/social.js]: Added URL parameters to the shortened links. See other comment.

File: kuma/wiki/templates/wiki/document.html [hunk 2]
```diff
@@ -334,6 +336,34 @@
                 {% endif %}
               </article>
 
+            {% if waffle.flag('share_links') %}
+                {% set full_url = request.build_absolute_uri(document.get_absolute_url()) %}
+
+                {% set share_url_twitter = full_url|urlparams(utm_campaign='share', utm_source='twitter.com', utm_medium='doc widget') %}
```
Comment (stephaniehobson) [inline kuma/wiki/templates/wiki/document.html]: @groovecoder I added the tracking parameters here :)

Comment (groovecoder) [inline kuma/wiki/templates/wiki/document.html]: :+1:

**Comment by groovecoder** [comment]
Looks good to me, are we calling this ready now?

**Comment by groovecoder** [comment]
Note: the Travis failure is the same one fixed in https://github.com/mozilla/kuma/pull/3120, so we can merge thru it.

**Comment by groovecoder** [comment]
Needs rebase after media/redesign PR merged.

**Comment by stephaniehobson** [comment]
The share text isn't #l10n friendly yet, maybe we should add logic that only shows the widget on English pages?

File: kuma/wiki/templates/wiki/document.html (hunk 2 above)
Comment (darkwing) [inline kuma/wiki/templates/wiki/document.html]: Maybe something other than 'doc widget', seems weird?  :D

**Comment by groovecoder** [comment]
Superseded by https://github.com/mozilla/kuma/pull/3124

**Pull request closed without merging**